foist/app.py

```diff
@@ -73,26 +85,43 @@ def dc_type(self):
         return DCTYPE.Text
 
     @property
+    def degree(self):
+        result = []
+        try:
+            degree = re.findall('[A-Z][a-z]{,4}\.? ?[A-Z][a-z]{,3}\.?[A-Z]?\.?[A-Z]?\.?[A-Z]?\.?', self.degree_statement)
+            for item in degree:
+                i = item.replace(' ', '')
+                i = i.rstrip('..')
```

Comment:
Just need one dot here `i.rstrip('.')`. It will remove as many trailing dots as it can.